.github/workflows/deploy_branches.yml

```diff
@@ -10,39 +10,67 @@ jobs:
       # and uncomment them to open an ssh connection at that point:
       #- name: Debugging with ssh
       #  uses: lhotari/action-upterm@v1
-      - name: Checkout the hazel repo on the current branch # STEP 1
+      - name: Checkout the hazel repo on the current branch
         uses: actions/checkout@v2
         with:
           path: source
       - name: Add the name of the current branch to the environment as BRANCH_NAME
         uses: nelonoel/branch-name@v1.0.1
-      - name: Retrieve the build environment if cached # STEP 2
+      - name: Retrieve the build environment if cached
         id: opam-cache
         uses: actions/cache@v2
         with:
           path: '/home/runner/.opam/'
           key: ${{ runner.os }}-modules-${{ hashFiles('./source/opam.export') }}
-      - name: Install dependencies and build hazel # STEP 3
+      - name: Set-up OCaml
         run: |
           sudo apt --assume-yes install curl m4 opam
           export OPAMYES=1
           opam init --compiler=ocaml-base-compiler.5.0.0
+      - name: Install dependencies
+        run: |
           eval $(opam env)
+          export OPAMYES=1
           make deps 
+        working-directory: ./source
+      - name: Build Hazel
+        run: |
+          eval $(opam env)
           make release
         working-directory: ./source
-      - name: Checkout the website build artifacts repo # STEP 4
+      - name: Run Tests
+        id: test
+        continue-on-error: true
+        run: |
+          eval $(opam env)
+          node _build/default/src/test/haz3ltest.bc.js > test_output
+          if [ $? -eq 0 ]; then
+            echo "::set-output name=tests_passed::true"
+          else
+            echo "::set-output name=tests_passed::false"
+          fi
+        working-directory: ./source
```

Comment:
I separated the steps so that we can track durations separately. When the whole installation was one step it was hard to see how much time was on dependencies vs the build itself.